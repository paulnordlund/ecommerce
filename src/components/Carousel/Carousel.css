.carousel-container {
    margin-bottom: 4em !important;
}

.main-item-detail-header {
    font-size: 1.5em;
    font-weight: 100;
    text-align: center;
}

.carousel-section {
    margin: 0 auto !important;
}

.carousel-image {
    width: 70% !important;
    margin: 0 auto !important;
    padding: 0 1.5em;
}

.carousel-thumbs {
    padding-top: 2em;
    width: 100%;
}