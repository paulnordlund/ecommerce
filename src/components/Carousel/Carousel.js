import React, { Component } from "react";
import { observer } from "mobx-react";
import Slider from "react-slick";
import { Row, Col } from "react-bootstrap";

@observer
export class Carousel extends Component {
  render() {
    const { title } = this.props;

    const settings = {
      // customPaging: function(i) {
      //   return (
      //     <a>
      //       <img
      //         className="carousel-thumbs"
      //         src={`http://target.scene7.com/is/image/Target/14263758_Alt0${i +
      //           1}`}
      //         alt=""
      //       />
      //     </a>
      //   );
      // },
      accessibility: true,
      arrows: false,
      className: "carousel-component",
      dots: true,
      dotsClass: "slick-dots slick-thumb",
      fade: true,
      infinite: true,
      lazyLoad: "ondemand",
      speed: 500,
      slidesToShow: 1,
      slidesToScroll: 1,
      swipe: true
    };

    const createMarkup = value => {
      return {
        __html: value
      };
    };

    return (
      <div className="carousel-container">
        <Row>
          <Col xs={12} className="title-section">
            <h2
              className="main-item-detail-header"
              dangerouslySetInnerHTML={createMarkup(title)}
            />
          </Col>
          <Col xs={12} className="carousel-section">
            <Slider {...settings}>
              <div><img className="carousel-image" src='//target.scene7.com/is/image/Target/GUEST_55b87cbe-a889-476b-b45d-e93abd41db53?wid=488&amp;hei=488&amp;fmt=webp' /></div>
              <div><img className="carousel-image" src='//target.scene7.com/is/image/Target/GUEST_9590cb47-0331-4736-9612-75f35968298f?wid=488&amp;hei=488&amp;fmt=webp' /></div>
              <div><img className="carousel-image" src='//target.scene7.com/is/image/Target/GUEST_98074548-376a-479a-85c7-748e53868981?wid=488&hei=488&fmt=webp' /></div>
              <div><img className="carousel-image" src='//target.scene7.com/is/image/Target/GUEST_5ddbeba3-3b3d-4675-b5fc-8e8337dc0e7a?fmt=webp&qlt=80&wid=1400' /></div>
            </Slider>
          </Col>
        </Row>
      </div>
    );
  }
}

export default Carousel;
