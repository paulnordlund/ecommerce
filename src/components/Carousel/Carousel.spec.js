import React from "react";
import { shallow } from "enzyme";
import Carousel from "./Carousel";

describe("Carousel", () => {
  it("renders", () => {
    const props = {
      title: "title",
    };

    const wrapper = shallow(<Carousel {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
