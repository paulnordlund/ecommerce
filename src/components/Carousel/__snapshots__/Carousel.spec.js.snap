// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Carousel
    title="title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Row
        bsClass="row"
        componentClass="div"
>
        <Col
                bsClass="col"
                className="title-section"
                componentClass="div"
                xs={12}
        >
                <h2
                        className="main-item-detail-header"
                        dangerouslySetInnerHTML={
                                Object {
                                        "__html": "title",
                                      }
                        }
                />
        </Col>
        <Col
                bsClass="col"
                className="carousel-section"
                componentClass="div"
                xs={12}
        >
                <Slider
                        accessibility={true}
                        arrows={false}
                        className="carousel-component"
                        customPaging={[Function]}
                        dots={true}
                        dotsClass="slick-dots slick-thumb"
                        fade={true}
                        infinite={true}
                        lazyLoad="ondemand"
                        slidesToScroll={1}
                        slidesToShow={1}
                        speed={500}
                        swipe={true}
                >
                        <div>
                                <img
                                        alt=""
                                        className="carousel-image"
                                        src="image"
                                />
                        </div>
                        <div>
                                <img
                                        alt=""
                                        className="carousel-image"
                                        src="image"
                                />
                        </div>
                </Slider>
        </Col>
</Row>,
      "className": "carousel-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "row",
        "children": Array [
          <Col
            bsClass="col"
            className="title-section"
            componentClass="div"
            xs={12}
>
            <h2
                        className="main-item-detail-header"
                        dangerouslySetInnerHTML={
                                    Object {
                                                "__html": "title",
                                              }
                        }
            />
</Col>,
          <Col
            bsClass="col"
            className="carousel-section"
            componentClass="div"
            xs={12}
>
            <Slider
                        accessibility={true}
                        arrows={false}
                        className="carousel-component"
                        customPaging={[Function]}
                        dots={true}
                        dotsClass="slick-dots slick-thumb"
                        fade={true}
                        infinite={true}
                        lazyLoad="ondemand"
                        slidesToScroll={1}
                        slidesToShow={1}
                        speed={500}
                        swipe={true}
            >
                        <div>
                                    <img
                                                alt=""
                                                className="carousel-image"
                                                src="image"
                                    />
                        </div>
                        <div>
                                    <img
                                                alt=""
                                                className="carousel-image"
                                                src="image"
                                    />
                        </div>
            </Slider>
</Col>,
        ],
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": <h2
              className="main-item-detail-header"
              dangerouslySetInnerHTML={
                            Object {
                                          "__html": "title",
                                        }
              }
/>,
            "className": "title-section",
            "componentClass": "div",
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "main-item-detail-header",
              "dangerouslySetInnerHTML": Object {
                "__html": "title",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": <Slider
              accessibility={true}
              arrows={false}
              className="carousel-component"
              customPaging={[Function]}
              dots={true}
              dotsClass="slick-dots slick-thumb"
              fade={true}
              infinite={true}
              lazyLoad="ondemand"
              slidesToScroll={1}
              slidesToShow={1}
              speed={500}
              swipe={true}
>
              <div>
                            <img
                                          alt=""
                                          className="carousel-image"
                                          src="image"
                            />
              </div>
              <div>
                            <img
                                          alt=""
                                          className="carousel-image"
                                          src="image"
                            />
              </div>
</Slider>,
            "className": "carousel-section",
            "componentClass": "div",
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessibility": true,
              "arrows": false,
              "children": Array [
                <div>
                  <img
                                    alt=""
                                    className="carousel-image"
                                    src="image"
                  />
</div>,
                <div>
                  <img
                                    alt=""
                                    className="carousel-image"
                                    src="image"
                  />
</div>,
              ],
              "className": "carousel-component",
              "customPaging": [Function],
              "dots": true,
              "dotsClass": "slick-dots slick-thumb",
              "fade": true,
              "infinite": true,
              "lazyLoad": "ondemand",
              "slidesToScroll": 1,
              "slidesToShow": 1,
              "speed": 500,
              "swipe": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt=""
                    className="carousel-image"
                    src="image"
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "carousel-image",
                    "src": "image",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt=""
                    className="carousel-image"
                    src="image"
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "carousel-image",
                    "src": "image",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Row
          bsClass="row"
          componentClass="div"
>
          <Col
                    bsClass="col"
                    className="title-section"
                    componentClass="div"
                    xs={12}
          >
                    <h2
                              className="main-item-detail-header"
                              dangerouslySetInnerHTML={
                                        Object {
                                                  "__html": "title",
                                                }
                              }
                    />
          </Col>
          <Col
                    bsClass="col"
                    className="carousel-section"
                    componentClass="div"
                    xs={12}
          >
                    <Slider
                              accessibility={true}
                              arrows={false}
                              className="carousel-component"
                              customPaging={[Function]}
                              dots={true}
                              dotsClass="slick-dots slick-thumb"
                              fade={true}
                              infinite={true}
                              lazyLoad="ondemand"
                              slidesToScroll={1}
                              slidesToShow={1}
                              speed={500}
                              swipe={true}
                    >
                              <div>
                                        <img
                                                  alt=""
                                                  className="carousel-image"
                                                  src="image"
                                        />
                              </div>
                              <div>
                                        <img
                                                  alt=""
                                                  className="carousel-image"
                                                  src="image"
                                        />
                              </div>
                    </Slider>
          </Col>
</Row>,
        "className": "carousel-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": Array [
            <Col
              bsClass="col"
              className="title-section"
              componentClass="div"
              xs={12}
>
              <h2
                            className="main-item-detail-header"
                            dangerouslySetInnerHTML={
                                          Object {
                                                        "__html": "title",
                                                      }
                            }
              />
</Col>,
            <Col
              bsClass="col"
              className="carousel-section"
              componentClass="div"
              xs={12}
>
              <Slider
                            accessibility={true}
                            arrows={false}
                            className="carousel-component"
                            customPaging={[Function]}
                            dots={true}
                            dotsClass="slick-dots slick-thumb"
                            fade={true}
                            infinite={true}
                            lazyLoad="ondemand"
                            slidesToScroll={1}
                            slidesToShow={1}
                            speed={500}
                            swipe={true}
              >
                            <div>
                                          <img
                                                        alt=""
                                                        className="carousel-image"
                                                        src="image"
                                          />
                            </div>
                            <div>
                                          <img
                                                        alt=""
                                                        className="carousel-image"
                                                        src="image"
                                          />
                            </div>
              </Slider>
</Col>,
          ],
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": <h2
                className="main-item-detail-header"
                dangerouslySetInnerHTML={
                                Object {
                                                "__html": "title",
                                              }
                }
/>,
              "className": "title-section",
              "componentClass": "div",
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "main-item-detail-header",
                "dangerouslySetInnerHTML": Object {
                  "__html": "title",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": <Slider
                accessibility={true}
                arrows={false}
                className="carousel-component"
                customPaging={[Function]}
                dots={true}
                dotsClass="slick-dots slick-thumb"
                fade={true}
                infinite={true}
                lazyLoad="ondemand"
                slidesToScroll={1}
                slidesToShow={1}
                speed={500}
                swipe={true}
>
                <div>
                                <img
                                                alt=""
                                                className="carousel-image"
                                                src="image"
                                />
                </div>
                <div>
                                <img
                                                alt=""
                                                className="carousel-image"
                                                src="image"
                                />
                </div>
</Slider>,
              "className": "carousel-section",
              "componentClass": "div",
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessibility": true,
                "arrows": false,
                "children": Array [
                  <div>
                    <img
                                        alt=""
                                        className="carousel-image"
                                        src="image"
                    />
</div>,
                  <div>
                    <img
                                        alt=""
                                        className="carousel-image"
                                        src="image"
                    />
</div>,
                ],
                "className": "carousel-component",
                "customPaging": [Function],
                "dots": true,
                "dotsClass": "slick-dots slick-thumb",
                "fade": true,
                "infinite": true,
                "lazyLoad": "ondemand",
                "slidesToScroll": 1,
                "slidesToShow": 1,
                "speed": 500,
                "swipe": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt=""
                      className="carousel-image"
                      src="image"
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "",
                      "className": "carousel-image",
                      "src": "image",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt=""
                      className="carousel-image"
                      src="image"
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "",
                      "className": "carousel-image",
                      "src": "image",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
